apis/multiarch/common/plugins/nodeaffinityscoring_plugin.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+const (
+	// PluginName for NodeAffinityScoring.
+	PluginName = "NodeAffinityScoringPluginName"
+)
+
+// Plugins represents the plugins configuration.
+type Plugins struct {
+	NodeAffinityScoring *NodeAffinityScoring `json:"nodeAffinityScoring,omitempty"`
+	// Future plugins can be added here.
+}
+
+// NodeAffinityScoring is the plugin that implements the ScorePlugin interface.
+type NodeAffinityScoring struct {
+	// Base is a required field for enabling/disabling the plugin.
+	Base PluginBase `json:"base" protobuf:"bytes,1,opt,name=base" kubebuilder:"validation:Required"`
+	// Args contains the configuration and is required.
+	Args *NodeAffinityScoringArgs `json:"args" protobuf:"bytes,2,opt,name=args" kubebuilder:"validation:Required"`
+}
+
+// PluginBase is a common base for all plugins, requiring the Enabled field.
+type PluginBase struct {
+	// Enabled is a required boolean field.
+	Enabled bool `json:"enabled" protobuf:"varint,1,opt,name=enabled" kubebuilder:"validation:Required"`
+}
+
+// PlatformConfig holds configuration for specific platforms, with required fields validated.
+type PlatformConfig struct {
+	// Architecture must be a non-empty string.
+	Architecture string `json:"architecture" protobuf:"bytes,1,rep,name=architecture" kubebuilder:"validation:Required"`
+	// Weight must be a non-negative integer.
+	Weight int `json:"weight" protobuf:"bytes,2,rep,name=weight" kubebuilder:"validation:Required,validation:Minimum=0"`
```

Comment:
are the kubebuilder tags verified?

We've always set them as comments above the field:

```suggestion
    // +kubebuilder:validation:Minimum:=0
    // +kubebuilder:validation:Required
	Weight int `json:"weight" protobuf:"bytes,2,rep,name=weight"`
```

(Look at similar ones in the code)